src/Stack/Sig/GPG.hs

```diff
@@ -0,0 +1,109 @@
+{-# LANGUAGE CPP             #-}
+{-# LANGUAGE RankNTypes      #-}
+{-# LANGUAGE RecordWildCards #-}
+
+{-|
+Module      : Stack.Sig.GPG
+Description : GPG Functions
+Copyright   : (c) FPComplete.com, 2015
+License     : BSD3
+Maintainer  : Tim Dysinger <tim@fpcomplete.com>
+Stability   : experimental
+Portability : POSIX
+-}
+
+module Stack.Sig.GPG (fullFingerprint, signPackage, verifyFile)
+       where
+
+#if __GLASGOW_HASKELL__ < 710
+import           Control.Applicative ((<$>))
+#endif
+
+import           Control.Monad.Catch (MonadThrow, throwM)
+import           Control.Monad.IO.Class (MonadIO, liftIO)
+import qualified Data.ByteString.Char8 as C
+import           Data.List (find)
+import           Data.Monoid ((<>))
+import qualified Data.Text as T
+import           Stack.Types
+import           System.Exit (ExitCode(..))
+import           System.Process (readProcessWithExitCode)
+
+-- | Extract the full long @fingerprint@ given a short (or long)
+-- @fingerprint@
+fullFingerprint
+    :: (Monad m, MonadIO m, MonadThrow m)
+    => Fingerprint -> m Fingerprint
+fullFingerprint (Fingerprint fp) = do
+    (code,out,err) <-
+        liftIO
+            (readProcessWithExitCode
```

Comment:
This should use the functions from System.Process.Read so that the modified environment is used, which will be useful especially for Windows.